### PR TITLE
ci: Added more templates

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] 🪄 Feature
- [ ] 🐞 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📚 Documentation
- [ ] 🧪 Test
- [x] 🛠️ Other

## Related issues

- Related issue # 
- Closes #

## Checklist

- [x] Code follows style guide
- [ ] Tests added
- [ ] Documented the changes

***

## Please explain your changes below.
- Changed a text area on the bug template from "place holder" to normal "value". The reason being that text was shown as a suggestion (in the text box) but it disappeared as the user started writing.  

- Changed the label from "bug" to "needs triage". This is because users may report a bug, but it may be just a misuse or misunderstanding of the library. Core maintainers need to confirm this, and set the proper label. (This would be like pandas workflow).  

- Added the possibility to open a blank issue if the topic of the templates is different.

- Added documentation, question and feature templates.

- Corrected a setting for codspeed.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -45,12 +46,11 @@ body:
 - type: textarea
   attributes:
     label: Please describe your system.
-    render: shell
-    placeholder: |
+    value: |
       1. OS and version: [e.g. Windows 10]
-      2. `sparse` version (`sparse.__version__`)
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply .github/ISSUE_TEMPLATE/bug_report.yml:51]: No

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+      - label: >
+           I checked that this issue has not been reported before
+           [list of issues](https://github.com/pydata/sparse/issues).
+        required: true
+      - label: >
+          I have confirmed this bug exists on the latest version of sparse.
+        required: true
+      - label: >
+            I have confirmed this bug exists on the main branch of sparse.
+        required: true
```
Comment (hameerabbasi) [inline .github/ISSUE_TEMPLATE/bug_report.yml:25]: Let's remove this requirement, as many users may not know how to install from git.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
One change and should be good to go.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (hameerabbasi) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: I meant let's keep the check box but remove the requirement.

Comment (DeaMariaLeon) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Ok

**Review by hameerabbasi — APPROVED** [review]

**Pull request merged**